### PR TITLE
avoid running CI for proposals or examples folders

### PR DESCRIPTION
**Pull request by casassg**
Avoid running CI for PRs under proposals or examples folders. This should avoid us running CI for cases that CI will ignore.

### CODE REVIEW

**Comment by casassg** [comment]
@rcrowe-google  can we please merge this? This contains a bunch of commits that got removed w the force push

File: .github/ci/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: ^examples/
```
Comment (TheMichaelHu) [inline .github/ci/.pre-commit-config.yaml:3]: Why exclude `examples/`?

Comment (casassg) [reply .github/ci/.pre-commit-config.yaml:3]: Running this on examples folder was failing, so I removed it until I have time to decide wether code under examples should be also run under the same CI as tfx_addons package.

I initially also scoped only for CI to work for tfx-addons. I'm open to

Comment (TheMichaelHu) [reply .github/ci/.pre-commit-config.yaml:3]: imo we should encourage good style + testing in examples. I'll take a look at why the tests were failing.

Comment (TheMichaelHu) [reply .github/ci/.pre-commit-config.yaml:3]: I see, the current CI assumes the root dir is at the project root while the tests in the scikit-learn example assume the root dir the scikit-learn example project dir.

Also, the checks from `pre-commit-checks` are highly opinionated, and I'm not sure they align with our internal style. Not sure if we plan to migrate any more code from the TFX codebase, but that might make it difficult.

Comment (casassg) [reply .github/ci/.pre-commit-config.yaml:3]: We can disable the license addition part and isort, Yapf and pylint are mirroring TFX repo CI. 

Would love to discuss and find a good middleground for this. I added some that were useful but agree that we shouldnt make CI too hard either

**Pull request merged**